src/TLuaInterpreter.cpp

```diff
@@ -17479,3 +17483,18 @@ int TLuaInterpreter::announce(lua_State *L) {
     mudlet::self()->announce(text, processing);
     return 0;
 }
+
+int TLuaInterpreter::setInboundSocketData(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    size_t len = 0;
+    const char* str = lua_tolstring(L, 1, &len);
+    
+    host.mSetInSocketData = true;
+    host.mSetInSocketDataBuffer.clear();
+    for (int i = 0; i < (int)len; i++) {
+        host.mSetInSocketDataBuffer.push_back(str[i]);
```

Comment:
You'll probably be getting a warning from one of the code review bots about indexing into raw c-strings...! :grinning:

Reply:
Ahh, OK.  What's the correct way to read a character at index?